### PR TITLE
Neat 239 a has data filter can reference at most 10 views and or containers

### PR DESCRIPTION
**Pull request by nikokaoja**
### Added
- Added schema validator for performance, specifically if views map to too many containers.

### CODE REVIEW

File: cognite/neat/rules/issues/dms.py [hunk 1]
```diff
@@ -250,6 +252,28 @@ def dump(self) -> dict[str, Any]:
         return output
 
 
+@dataclass(frozen=True)
+class ViewMapsToTooManyContainersWarning(DMSSchemaWarning):
+    description = "The view maps to more than 9 containers which impacts read/write performance of data model"
```
Comment (nikokaoja) [inline cognite/neat/rules/issues/dms.py:257]:
```suggestion
    description = "The view maps to more than 10 containers which impacts read/write performance of data model"
```

File: cognite/neat/rules/issues/dms.py [hunk 2]
```diff
@@ -442,6 +466,28 @@ def dump(self) -> dict[str, Any]:
         return output
 
 
+@dataclass(frozen=True)
+class HasDataFilterAppliedToTooManyContainersWarning(DMSSchemaWarning):
+    description = "The view filter hasData applied to more than 10 containers this will cause DMS API Error"
+    fix = "Do not map to more than 10 containers, alternatively to override the filter by using rawFilter"
```
Comment (nikokaoja) [inline cognite/neat/rules/issues/dms.py:472]:
```suggestion
    fix = "Do not map to more than 10 containers, alternatively override the filter by using rawFilter"
```

File: cognite/neat/rules/models/dms/_schema.py [hunk 3]
```diff
@@ -60,6 +62,30 @@ class DMSSchema:
...
+        directly_referenced_containers = indexed_views[view_id].referenced_containers()
+        inherited_referenced_containers = set()
+
+        for view_id in view_inheritance:
+            if implemented_view := indexed_views.get(view_id):
+                inherited_referenced_containers |= implemented_view.referenced_containers()
+            else:
+                raise IncompleteSchemaError(missing_component=view_id).as_exception()
+
+        return directly_referenced_containers | inherited_referenced_containers
```
Comment (doctrino) [inline cognite/neat/rules/models/dms/_schema.py:87]: Thanks this helped, I struggled with the last one.

File: cognite/neat/rules/models/dms/_schema.py [hunk 4]
```diff
@@ -60,6 +62,30 @@ class DMSSchema:
         "node": "node_types",
     }
 
+    def _get_mapped_container_from_view(self, view_id: dm.ViewId) -> set[dm.ContainerId]:
+        # index all views, including ones from reference
+        indexed_views = {
+            **{view.as_id(): view for view in self.views},
+            **({view.as_id(): view for view in self.reference.views} if self.reference else {}),
+        }
```
Comment (doctrino) [inline cognite/neat/rules/models/dms/_schema.py:70]: Suggest adding this as a cached_property as it is independent of the method input.

**Review by doctrino — APPROVED** [review]
One suggestion.

File: cognite/neat/rules/models/dms/_schema.py (hunk 4 above)
Comment (doctrino) [reply cognite/neat/rules/models/dms/_schema.py:70]: Nevermind, that will cause issues if more views are added later.

**Pull request merged**